core/coreapi/interface/interface.go

```diff
@@ -25,6 +28,11 @@ type Path interface {
 type Node ipld.Node
 type Link ipld.Link
 
+type IpnsEntry struct {
+	Name  string
+	Value Path
+}
```

Comment:
I wonder if we should make `IpnsEntry` an interface, in the spirit of future extensibility. Opinions on that?

Reply:
On the topic of future extensibility, we should also consider adding TTL/expiry to the returned values.